api/app/controllers/webhook.controller.ts

```diff
@@ -154,3 +156,100 @@ export async function create(req: Request, res: Response) {
     }
   }
 }
+
+export async function handlePushEvent(req: Request, res: Response) {
+  if (!verify_signature(req)) {
+    return res.status(401).json("Unauthorized");
+  }
+
+  const pushEvent = req.body;
+  if (!pushEvent) {
+    return res.status(500).json({
+      message: "No push event found in the request body.",
+    });
+  }
+
+  const commits = pushEvent.commits;
+  if (!commits) {
+    return res.status(500).json({
+      message: "No commits found in the request body.",
+    });
+  }
+
+  let responseStatus = 200;
+  let responseMessage = "success";
+
+  try {
+    for (const commit of commits) {
+      const changedFiles = [...commit.added];
+      for (const file of changedFiles) {
+        const rawUrl = `https://raw.githubusercontent.com/${pushEvent.repository.full_name}/master/${file}`;
+        const response: AxiosResponse<TSTBTCAttributes> = await axios.get(
+          rawUrl
+        );
+        const mdContent = response.data;
+        const jsonContent = parseMdToJSON(mdContent);
+        // Validate jsonContent
+        if (!jsonContent) {
+          responseStatus = 400;
+          responseMessage =
+            "Malformed data: transcript content might not be in the correct format";
+          break;
+        }
+        const transcriptHash = generateUniqueHash(jsonContent);
+        const totalWords = getTotalWords(jsonContent.body);
+        const content = jsonContent;
+        const originalContent = jsonContent;
+
+        // Validate other values
+        if (!transcriptHash || !totalWords) {
+          responseStatus = 400;
+          responseMessage =
+            "Malformed data: transcript content might not be in the correct format";
+          break;
+        }
+
+        const existingTranscript = await Transcript.findOne({
+          where: { transcriptHash: transcriptHash },
+        });
+
+        if (existingTranscript) {
+          responseStatus = 409;
+          responseMessage = "transcript already exists";
+          break;
+        }
+
+        if (
+          jsonContent.transcript_by.includes("TSTBTC") &&
```

Comment:
are we sure this (TSTBTC) always comes in uppercase? we should probably also check for lowercase.
would've be nice if we could normailze the text before the `include` but I don't see how.

Reply:
Yes, [this is the only place we define it](https://github.com/bitcointranscripts/tstbtc/blob/a0a4d9f49e37c2a6e851966be9ace7330c618f59/app/transcription.py#L348).

Reply:
we can just convert the transcript_by key values to lowercase instead(what ever is coming in we read in lowercase)